src/sql/services/notebook/notebookRegistry.ts

```diff
@@ -0,0 +1,116 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+'use strict';
+
+import { IJSONSchema } from 'vs/base/common/jsonSchema';
+import { ExtensionsRegistry, IExtensionPointUser } from 'vs/workbench/services/extensions/common/extensionsRegistry';
+import { localize } from 'vs/nls';
+import * as platform from 'vs/platform/registry/common/platform';
+
+export const Extensions = {
+	NotebookProviderContribution: 'notebook.providers'
+};
+
+export interface NotebookProviderDescription {
+	provider: string;
+	fileTypes: string | string[];
+}
+
+let notebookProviderType: IJSONSchema = {
+	type: 'object',
+	default: { provider: '', fileTypes: [] },
+	properties: {
+		provider: {
+			description: localize('carbon.extension.contributes.notebook.provider', 'Identifier of the notebook provider.'),
+			type: 'string'
+		},
+		fileTypes: {
+			description: localize('carbon.extension.contributes.notebook.fileTypes', 'What types of file should be registered to this notebook provider'),
```

Comment:
maybe we can say "file extensions" instead of "types of file"

Reply:
Yeah this is interesting since for untitled files, it's more the "language mode" but yeah I agree.